mono/tests/bug-8477.il

```diff
@@ -0,0 +1,115 @@
+// This test is meant to make sure Mono doesn't fail when invalid COM invocations are present but never executed.
+// See https://bugzilla.xamarin.com/show_bug.cgi?id=8477 for details.
```

Comment:
Why does this needs to be an IL test?

Reply:
The original bug report provides two assemblies, an exe and a dll. I figured the most transparent way of making a test would be converting it into the IL form, cleaning it up a little and merging together into a single IL file. What test format would you suggest that I use?

Reply:
C#

Reply:
Done, though to be honest, I think rewriting an IL test case in C# has a chance of defeating its initial purpose. Newer versions of csc/mcs may change the way they handle some edge cases in the future , thus producing assemblies that won't check for what they were meant to check in the first place. For instance, they could "optimize too much" and eliminate a branch that is never taken but needs to be there to trigger the bug, or detect an invalid COM type during compilation.